docs/source/implementations/portfolio_optimisation.rst

```diff
@@ -221,6 +221,118 @@ Implementation
 
 
 
+Risk Estimators
+===============
+
+This class includes de-noising algorithms and helping methods. In particular, methods for de-noising the covariance matrix and transforming the covariance matrix to the correlation matrix and back to the covariance matrix.
+
+The main idea behind de-noising is to separate the noise-related eigenvalues from the signal-related ones. This is achieved through fitting the Marcenko-Pastur distribution of the empirical distribution of eigenvalues using a Kernel Density Estimate (KDE).
+
+This method is described in more detail in the NCO Notebook.
+
+Implementation
+~~~~~~~~~~~~~~
+
+.. automodule:: mlfinlab.portfolio_optimization.risk_estimators
+
+    .. autoclass:: RiskEstimators
+        :members:
+
+        .. automethod:: __init__
+
+
+
+
+Nested Clustered Optimization (NCO)
+===================================
+
+The Nested Clustered Optimization algorithm estimates optimal weights allocation to either maximize the Sharpe ratio
+or minimize the variance of a portfolio.
+
+The steps of the NCO algorithm are:
+
+1. Get the covariance matrix of the outcomes as an input (and the vector of means if the target is to maximize the Sharpe ratio).
+
+2. Transform the covariance matrix to the correlation matrix and calculate the distance matrix based on it.
+
+3. Cluster the covariance matrix into subsets of highly-correlated variables.
+
+4. Compute the optimal weights allocation for every cluster.
+
+5. Reduce the original covariance matrix to a reduced one - where each cluster is represented by a single variable.
+
+6. Compute the optimal weights allocation for the reduced covariance matrix.
+
+7. Compute the final allocations as a dot-product of the allocations between the clusters and inside the clusters.
+
+
+This algorithm is described in more detail in the work **A Robust Estimator of the Efficient Frontier** *by* Marcos Lopez de Prado `available here <https://papers.ssrn.com/abstract_id=3469961>`_.
+
+Implementation
+~~~~~~~~~~~~~~
+
+.. automodule:: mlfinlab.portfolio_optimization.nco
+
+    .. autoclass:: NCO
+        :members: __init__, opt_port_nco, opt_port
+
+
+
+
+Monte Carlo Optimization Selection (MCOS)
+=========================================
+
+The Monte Carlo Optimization Selection algorithm calculates the NCO allocations and a simple optimal allocation for multiple simulated pairs of mean vector and the covariance matrix to determine the most robust method for weights allocations for a given pair of means vector and a covariance vector.
```

Comment:
change "weights allocations" to "weight allocations"

Reply:
Done